errdef/errors.go

```diff
@@ -20,11 +20,12 @@ import "errors"
 // Common errors used in ORAS
 var (
 	ErrAlreadyExists      = errors.New("already exists")
-	ErrUnsupported        = errors.New("unsupported")
-	ErrInvalidReference   = errors.New("invalid reference")
+	ErrBadRequest         = errors.New("bad request")
```

Comment:
The problem is that we can't tell the reason of the 400 response, it can be because the manifest media type is not supported, or the manifest format is invalid. What if some users push an artifact manifest in an invalid form, and the server returns 400 while it actually supports artifact manifest?

Reply:
Mmmm, it may not be a problem after we implement referrers compatibility for manifest push, as the content will be parsed before being pushed anyway.